src/neo/SmartContract/Native/LedgerContract.cs

```diff
@@ -126,7 +132,7 @@ public bool ContainsBlock(DataCache snapshot, UInt256 hash)
         /// <returns><see langword="true"/> if the blockchain contains the transaction; otherwise, <see langword="false"/>.</returns>
         public bool ContainsTransaction(DataCache snapshot, UInt256 hash)
         {
-            return snapshot.Contains(CreateStorageKey(Prefix_Transaction).Add(hash));
+            return snapshot.Contains(CreateStorageKey(Prefix_Transaction).Add(hash)) || snapshot.Contains(CreateStorageKey(Prefix_TrimmedTransaction).Add(hash));
```

Comment:
That's two requests to the DB.

Reply:
The logic here is quite contradictory. There are two options for persisting conflicting transactions. One is to record the hash of the conflicting transaction separately, which leads to the need to query twice.
The other is to store with other transactions. But it actually saves an empty transaction, which will have an impact on the operation of the contract to obtain the transaction.
The reason why I choose the first one here is that this is an internal method, and the scope of influence is relatively small

Reply:
Correct me if I'm wrong, but this method is what's going to be used for transaction (non-)existence check, so this is likely to have some influence on benchmarks. It's logically correct, so if you're sure it's OK performance-wise then I'm perfectly fine with it.

Reply:
I see. Depending on the characteristics of levelDB, it may have an impact, so I am also tangled. Because there is some interoperable services,such as  GetTransactionState() in LedgerContract cannot distinguish between empty transactions and normal transactions, unless additional flags are added in TransactionState, which will cause changes to the storage area.

Reply:
https://github.com/neo-ngd/neo/pull/7